File: twosum.js
var twoSum = function(nums, target) {
    let holder = {};
    
    for (let i = 0; i < nums.length; i++){
        console.log(holder)
        let rTarget = target - nums[i];
        if (rTarget in holder){
            return [holder[rTarget], i];
        }
        holder[nums[i]] = i;
    }
};


let a = [3,2,4,5,3,2,10];
let b = 12;
console.log(twoSum(a,b))

File: twoSum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for index, value in enumerate(nums):
            a = target - value
            if a in d:
                return d[a], index
            d[value] = index
        return d


nums = [2,3,4,5]
target = 9
a = Solution()
print(a.twoSum(nums,target))

